test/mpi/threads/init/mult_session.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
```

Comment:
I understand that you want to keep the test as close as to the original example. But all the tests check only `WORLD` pset. Can we extend the tests to also cover `SELF` pset?